src/plugins/intel_cpu/src/dnnl_extension_utils.cpp

```diff
@@ -150,10 +153,17 @@ std::shared_ptr<DnnlBlockedMemoryDesc> DnnlExtensionUtils::makeUndefinedDesc(con
 
 DnnlMemoryDescPtr DnnlExtensionUtils::query_md(const const_dnnl_primitive_desc_t& pd, const dnnl::query& what, int idx) {
     auto query = dnnl::convert_to_c(what);
-    const dnnl_memory_desc_t* cdesc = dnnl_primitive_desc_query_md(pd, query, idx);
+    const auto* cdesc = dnnl_primitive_desc_query_md(pd, query, idx);
+
     if (!cdesc)
         IE_THROW() << "query_md failed for query=" << query << " idx=" << idx << ".";
-    return DnnlExtensionUtils::makeDescriptor(*cdesc);
+
+    dnnl_memory_desc_t cloned_md = nullptr;
+    dnnl_memory_desc_clone(&cloned_md, cdesc);
```

Comment:
Since 3.0 oneDNN's C++ api structures are only holding the references (actually the pointers) to underlying C api structures.
So every time we want to copy desc we have to perform deep copy and in this case we need to clone underlying C api structure using heap (there is "new" call inside this clone) and the wrap it into memory::desc, so this memory::desc is responsible for de-allocation now.
I have also added a copy constructor for memory::desc itself which is basically doing the same (deep copy).

Reply:
As our measurements showed - such kind of additional clone in query routine lead to performance overheads most damaging dynamic shapes performance. We need to find solution which will allow to avoid deep copy of md here. However that can be done in follow up PR.

Reply:
After more investigation it turns out we can avoid this extra clone.
We just need to stick with C dnnl_memory_desc structure and pass it though until we assign it within our MemoryDesc wrapper classes.